app/Party.php

```diff
@@ -382,10 +383,16 @@ public function attendees()
 
     public function scopeUpcomingEvents($query, $by_event = false)
     {
+      // We want to show approved events (wordpress_post_id IS NOT NULL) or where we are a host (e.g. because
+      // we created it.
       if( $by_event ) {
         return $this->join('groups', 'groups.idgroups', '=', 'events.group')
                      ->join('events_users', 'events_users.event', '=', 'events.idevents')
-                     ->whereNotNull('events.wordpress_post_id')
+                     ->where(function($query) {
+                         $query
+                         ->whereNotNull('events.wordpress_post_id')
+                         ->orWhere('events_users.role', '=', Role::HOST);
```

Comment:
I'm not sure how this is working.  It looks like it is saying:

- if the event has been approved, then return it
- or if the event has a host, return it

It doesn't seem to be restricting to the query to any particular host, just saying as long as the event has one, pull it back.  Is that right?  (It might well be!)

I'm not sure what the $by_event flag refers to, so I don't know where this is being called.

Reply:
I remember looking at this, but I can't remember what I thought.  I can't find a use of this by_event flag at all, though I might not have got my head round scoped stuff properly.  Can you see if you can?

Reply:
To confirm, I also can't see anywhere that the upcomingEvents scope is being used and is being passed a value for the by_events flag.